### ISSUE TITLE
None of the bonus features work for me?

### ISSUE THREAD

**Issue by Kingdingbat** [kangarko/ChatControl-Pro#561]
Hi there!

The plugin is working great, except for all of the bonus features are not working. Am I missing something?
- @kangarko  tagging doesn't work
- I tried adding a rule  to filter console messages and it didn't work. (Copied and pasted from a known working rule on the forums)
- Grammar and punctuation correction doesn't work.
- I haven't messed with Variables, but I'd bet it doesn't work either.

- I have the latest of everything, including ProtocolLib
- I have granted the permission chatcontrol.notify.whenmentioned to to all the players
- I've searched the wiki, and all the help links I can find
- Ive checked to make sure the config is enabled multiple times
- I don't see errors in the console, or the plugin's log.

The rest of the features are all working great - This leads me to believe I've missed something important. Got any ideas?

**Comment by kangarko** [comment]
Post me your settings.yml to pastebin.com.

Also post me the console filter rules. It works for me. Not all messages can be matched, though.

> Grammar and punctuation correction doesn't work.
OPs have this disabled by default, test on player account. Check Wiki permissions pages.

> I haven't messed with Variables, but I'd bet it doesn't work either.
Well sorry but you have to test it first before claiming it does not work.

**Comment by Kingdingbat** [comment]
I didn't claim it doesn't work, I said I suspect it doesn't, because none of the other related (bonus) features are working. I also said that I suspect *I* have missed something important.

Here is a pastebin of my settings:
https://pastebin.com/rgAGp13E

The console filter rule I am trying to use is:
`#Hide automatic /list command from console
match CONSOLE issued server command: /list
dont verbose
then deny`

Since installing EssentialsX the console spams "CONSOLE issued server command /list" every 30 seconds. I'm trying to hide that. This rule was found as a solution to the problem on another support thread elsewhere and they said it worked.

Grammar and punctuation is not working for any player on the server. I have tested it myself on an alternate non-op account, and asked the rest of the players and none of them get corrected when they type something that should be corrected. 

I didn't CLAIM variables was broken, I said I suspect they aren't since the other 3 related (bonus) features are all not working either... they are all 4 bonus features and so far are the only thing on the plugin that seem to be not working, so perhaps there is something they all rely on that I am missing.

**Comment by kangarko** [comment]
I just tested the tagging feature and it is working for me. Make sure you test it from one account to another, as you cannot tag yourself, and you also include the @.
![a](https://i.imgur.com/Prdv0ah.png)

This is not how the rules work. Where do you put that rule? We are not PwnFilter in case you are coming from there, and do not have console.txt file. The file commands.txt is for in-game / commands. Use the Console_Filter section in settings.yml and add the message in there:

![b](https://i.imgur.com/8yg6lQz.png)

If grammar and punctuation does not work for anyone, check if you have enabled Grammar section in settings.yml. It works from my testing fully as expected. Also, it may be another plugin blocking it. Try reading this article and playing with the Listener_Priority.Checker: https://github.com/kangarko/ChatControl-Pro/wiki/Listener-Priorities

![e](https://i.imgur.com/liYRWuK.png)
![d](https://i.imgur.com/4PyAf4a.png)

**Comment by Kingdingbat** [comment]
Did you even bother to read the pastbin I took the time to give you, like you asked? If you had, you'd not even ask if I have grammar enabled. Why ask for that if you're not going to even make an effort to help?

I'm not some "n00b" that is just throwing random crap at you asking you to help me fix stupid mistakes. I've been working in the gaming industry, starting with QA since before you were born. I am very aware about how things work differently betwen Op/Admin account and players and I would not even have submitted this issue without testing on multiple installs, multiple accounts and verified that none of the other players can use it either. I wouldn't even come to you before spending hours upon hours pouring over your documentation and the settings files. So forgive me that I am done with your shitty attitude toward your PAYING customers.

Your section about rules is so filled with contempt that it barely makes any sense. In fact, your very first response to me was so irritated and filled with "Its not my problem, its your problem" attitude that I am sorry I spent money on you. In contrast, I repeatedly indicated that this was probably my mistake as it was, as to NOT imply there was a problem with your work, but I doubt that's true, now.

I'm not going to bother demanding a refund because it was my own idiocy that assumed that even though I'd have to wait for your personal life before I get help for any issues I may have after I buy your plugin, that I might actually be met with professionalism and gratitude that I felt confident in your work that I could pay for it.

So, now I regret giving you my money and I will probably never buy another plugin because of your shitty attitude. Great job!
